### PR TITLE
feat: add capability of add custom metaComponents

### PR DESCRIPTION
**Pull request by chilingling**
# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
  - Introduced `MetaInput` component for customizable input fields.
  - Added `MyInputConfigurator` for configuring input settings.
  - Introduced `MetaSelect` component for customizable select dropdowns.
  - Added `InputConfigurator` and `SelectConfigurator` for property panel customization.

- **Updates**
  - Renamed components in various configurations from "MetaInput" to "InputConfigurator" and "MetaSelect" to "SelectConfigurator".
  - Updated initialization scripts to register new meta components and configurators.

- **Documentation**
  - Added README files for `meta-components` and `configurator` packages.

### CODE REVIEW

File: packages/common/component/ConfigItem.vue [hunk 1]
```diff
@@ -201,13 +202,20 @@ export default {
...
-        : props.metaComponents[widget.value.component] ||
-          MetaComponents[widget.value.component] ||
-          MetaComponents['MetaInput']
-    )
+    const component = computed(() => {
+      if (multiType.value) {
+        return MultiTypeSelector
+      }
+
+      const component = getMetaComponent(widget.value.component)
```
Comment (zzcr) [inline packages/common/component/ConfigItem.vue]: 这块属性面板对应的Meta组件之前评审说是要放到生态中心，所以这里不应该和注册表有耦合吧？

File: designer-demo/src/metaComponents.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import CustomMetaInput from './MetaInput.vue'
+import { MetaInput, MetaSelect } from '@opentiny/tiny-engine-meta-components'
+
+export const metaComponents = [
+  {
+    name: 'meta-input-custom',
+    component: CustomMetaInput
+  },
+  MetaInput,
+  MetaSelect
```
[the remark this answers is not in the document]
Comment (chilingling) [reply designer-demo/src/metaComponents.js]: @coderabbitai `MetaInput` & `MetaSelect` already is the structure of  `name` and `component`

**Review by zzcr — DISMISSED** [review]

File: packages/meta-components/vite.config.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * Use of this source code is governed by an MIT-style license.
+ *
+ * THE OPEN SOURCE SOFTWARE IN THIS PRODUCT IS DISTRIBUTED IN THE HOPE THAT IT WILL BE USEFUL,
+ * BUT WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS FOR
+ * A PARTICULAR PURPOSE. SEE THE APPLICABLE LICENSES FOR MORE DETAILS.
+ *
+ */
+
+import { defineConfig } from 'vite'
+import path from 'path'
```
[the remark this answers is not in the document]
Comment (chilingling) [reply packages/meta-components/vite.config.js]: @coderabbitai why `import path form 'node:path'` is better than `import path from 'path'`

**Review by wenmine — COMMENTED** [review]
根据讨论，需要区分配置组件和meta(元)的概念，避免混淆，要把 meta -> configurator

File: packages/design-core/src/registerMetaComponents.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// TODO: 抽空 common package 中的 metaComponent
+import { MetaComponents } from '@opentiny/tiny-engine-common'
+import { addMetaComponents } from '@opentiny/tiny-engine-entry'
+
+/**
+ * 注册TinyEngine默认的 metaComponents
+ */
+export const registerMetaComponents = (metaComponents) => {
+  const { MetaInput, MetaSelect, ...otherComponents } = MetaComponents
```
Comment (hexqi) [inline packages/design-core/src/registerMetaComponents.js]: 为什么这里要把MetaInput和MetaSelect单独拎出来

Comment (chilingling) [reply packages/design-core/src/registerMetaComponents.js]: 这里当前是为了能够正常渲染，整体完成迁移之后会删除。不再注册 `MetaComponents`

File: packages/common/component/ConfigItem.vue [hunk 5]
```diff
@@ -201,13 +202,14 @@ export default {
...
+      // TODO: 需要弄清楚 props.metaComponents[widget.value.component] 是什么场景
+      return multiType.value
         ? MultiTypeSelector
-        : props.metaComponents[widget.value.component] ||
-          MetaComponents[widget.value.component] ||
-          MetaComponents['MetaInput']
-    )
+        : getMetaComponent(widget.value.component) ||
+            props.metaComponents[widget.value.component] ||
+            getMetaComponent('MetaInput')
```
Comment (wenmine) [inline packages/common/component/ConfigItem.vue]: 这个MetaInput也需要修改一下~

File: designer-demo/src/MyInputConfigurator.vue [hunk 6]
```diff
@@ -0,0 +1,63 @@
+<template>
+  <span>我是自定义的 meta input</span>
```
Comment (wenmine) [inline designer-demo/src/MyInputConfigurator.vue]: 这里的meta也顺便调整一下吧

File: packages/configurator/README.md [hunk 7]
```diff
@@ -0,0 +1,3 @@
+# TinyEngine 的 meta component 组件
```
Comment (wenmine) [inline packages/configurator/README.md]: 这里的meta也一起修改一下

File: packages/design-core/src/registerMetaComponents.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+// TODO: 抽空 common package 中的 metaComponent
+import { MetaComponents } from '@opentiny/tiny-engine-common'
+import { addMetaComponents } from '@opentiny/tiny-engine-entry'
+
+/**
+ * 注册TinyEngine默认的 metaComponents
+ */
+export const registerMetaComponents = (metaComponents) => {
```
Comment (wenmine) [inline packages/design-core/src/registerMetaComponents.js]: 同样的，addMetaComponents，registerMetaComponents，getMetaComponent是不是也要跟随一起修改比较好？

File: packages/entry/README.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
+# entry(暂定)
+
```
Comment (wenmine) [inline packages/entry/README.md:2]: readme里的也要统一调整

**Review by wenmine — DISMISSED** [review]

File: packages/common/component/ConfigItem.vue [hunk 10]
```diff
@@ -201,13 +202,14 @@ export default {
...
+    const component = computed(() => {
+      // TODO: 需要弄清楚 props.metaComponents[widget.value.component] 是什么场景
+      return multiType.value
         ? MultiTypeSelector
-        : props.metaComponents[widget.value.component] ||
-          MetaComponents[widget.value.component] ||
-          MetaComponents['MetaInput']
-    )
+        : getConfigurator(widget.value.component) ||
+            props.metaComponents[widget.value.component] ||
```
Comment (zzcr) [inline packages/common/component/ConfigItem.vue:210]: 这个地方叫metaComponents是不是不太合适？叫configuratorComponents

File: packages/design-core/src/registerConfigurators.js [hunk 11]
```diff
@@ -0,0 +1,11 @@
+// TODO: 抽空 common package 中的 metaComponent
```
[the remark this answers is not in the document]
Comment (chilingling) [reply packages/design-core/src/registerConfigurators.js:1]: @coderabbitai thanks fot that. remain as todo is just fine now.

File: packages/entry/src/configurators.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
+const configuratorsMap = new Map()
+
+export const addConfigurator = (components) => {
+  if (Array.isArray(components)) {
+    for (const { name, component } of components) {
+      configuratorsMap.set(name, component)
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (chilingling) [reply packages/entry/src/configurators.js]: @coderabbitai good idea, but i will remain as todo now.

File: packages/common/component/ConfigItem.vue (hunk 10 above)
Comment (chilingling) [reply packages/common/component/ConfigItem.vue:210]: 这个地方先留着，需要先看看透传 metaComponent 的场景是啥。

**Review by zzcr — APPROVED** [review]

**Pull request merged**